doc/design_AsyncNDRegisterAcessor_and_NumericAddressedBackend.dox

```diff
@@ -0,0 +1,60 @@
+
+The data transport queue is implmented as continuation of the notification queue, such that the read operations of the notification queue trigger the filling of the user buffer.
+
+The AsyncNDRegisterAccessor is ready to use. It provides a generic asynchronous implementation.
+Each backend has to store a list of accessors and call AsyncNDRegisterAccessor::activate(), AsyncNDRegisterAccessor::deactivate(), AsyncNDRegisterAccessor::sendDestructively() and AsyncNDRegisterAccessor::sendException() appropriately. As the way the data containers are organised and the asynchronous mechanism is implemented stongly depend on the backend, the implementation of these are left to the specific backend.
+However, there is a common interface implemented to allow the clean subscription/unsubscription from the backend. The AsyncNDRegisterAccessor is templated to an AsyncAccessorManager and an AsyncAccessorID, which are both handed over in the constructor. In its destructor the AsyncNDRegisterAccessor is calling
+<pre>
+void AsyncAccessorManager::unsubscribe(AsyncAccessorID id);
+</pre>
+This function has to be implemented by the manager. From the ID it knows which subscriber is to be removed. For this mechanism to work, the AsyncAccessorManager must store boost::weak_pointers and lock them when the instance is being accessed. Otherwise the one instance held within the manager prevents the destructor from being called.
+
+\section  design_async_NumericAddressedBackend Asynchronous accessors in the NumericAddressedBackend
+
+As the NumericAddressedBackend base class already provides the creation of the synchronous NumericAddressedBackendRegisterAccessors, it also provides the subscription mechanism for the asynchronous accessors. The NumericAddressedBackend is based on a map file which allows to describe registers with interrupts. In the interface to the hardware, all interrupts are void. The map file then descibes which of the synchronous registers are read out and pushed when the according interrupt is triggered. This mechanism is implemented in the NumericAddressedBackend base class and the protocol implementing backends only have to notify the NumericAddressedInterruptDispatcher that the interrupt has happened.
+
+\subsection  async_NumericAddressedBackend_interface Interface for backend implementations
+
+The NumericAddressedInterruptDispatcher serves two purposes:
+\li It serves as a subscription manager for the AsyncNDRegisterAccessor
+\li It provides functions which act on all subsribers.
```

Comment:
Doesn't it also handle the readouts through the synchronous accessors and passing on the (copied) data to the async accessors?

Reply:
Yes, that's one of the functions that acts on all subscibers. I guess I should re-phrase that.